### PR TITLE
Update-DG

### PR DESCRIPTION
**Pull request by oLiXino**

### CODE REVIEW

File: docs/diagrams/ModelClassDiagram.puml [hunk 1]
```diff
@@ -5,52 +5,60 @@ skinparam arrowColor MODEL_COLOR
...
-Name -[hidden]right-> Phone
-Phone -[hidden]right-> Address
-Address -[hidden]right-> Email
+Library *--> "1" UniqueDeckList
+GameManager *--> "1" UniqueCardList
+UniqueDeckList o--> "*" Deck
+Deck *--> Name
+Deck *--> "1" UniqueCardList
+UniqueCardList o--> "*" Card
+Card *--> FrontFace
```
Comment (Amoscheong97) [inline docs/diagrams/ModelClassDiagram.puml:58]: The Multiplicity of the FrontFace and BackFace should be 1.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -122,21 +122,22 @@ image::ModelClassDiagram.png[]
...
-* stores the Address Book data.
-* exposes an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
+* stores the Library data.
+* stores a `GameManager` object that represents one game session.
+* stores a `Deck` object that represents the deck that the user is viewing when user is in deck view.
+* stores a `Card` object that represents the card that the user is playing with when user is in play mode.
+* stores a `View` object that represents the view that the user is currently in.
+* stores a `Mode` object that represents the mode that the user is currently in.
+* exposes an unmodifiable `ObservableList<Deck>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
 * does not depend on any of the other three components.
```
Comment (Cherweijie) [inline docs/DeveloperGuide.adoc:132]: In Figure 7, I think you can improve on it by showing the multiplicities of the Front Face, Back Face of each card and also between Deck and Name, etc.

**Review by Cherweijie — COMMENTED** [review]
Overall, well done for adding and modifying the UML diagrams. I like that your app is unique and it is something I haven't seen before thus far in this module. I think you could improve on your DG by proofreading it perhaps once or twice more to look at some parts that you missed out on modifying. My group spotted some parts that were still elements of the AddressBook.

**Review by Amoscheong97 — COMMENTED** [review]
All in all, it is good. Just have to make the necessary changes on the errors we found.

**Pull request merged**